pkg/services/ngalert/alert_query.go

```diff
@@ -0,0 +1,250 @@
+package ngalert
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+const defaultMaxDataPoints float64 = 100
+const defaultIntervalMS float64 = 1000
+const defaultExprDatasourceID = -100
+
+type duration time.Duration
+
+func (d duration) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(d).Seconds())
+}
+
+func (d *duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+	switch value := v.(type) {
+	case float64:
+		*d = duration(time.Duration(value) * time.Second)
+		return nil
+	default:
+		return fmt.Errorf("invalid duration %v", v)
+	}
+}
+
+// RelativeTimeRange is the per query start and end time
+// for requests.
+type RelativeTimeRange struct {
+	From duration
+	To   duration
+}
+
+// IsValid checks that From duration is greater than To duration.
+func (rtr *RelativeTimeRange) IsValid() bool {
+	return rtr.From > rtr.To
+}
+
+func (rtr *RelativeTimeRange) toTimeRange(now time.Time) backend.TimeRange {
+	return backend.TimeRange{
+		From: now.Add(-time.Duration(rtr.From)),
+		To:   now.Add(-time.Duration(rtr.To)),
+	}
+}
+
+// AlertQuery represents a single query associated with an alert definition.
+type AlertQuery struct {
```

Comment:
Doesn't have to be public:
```suggestion
// alertQuery represents a single query associated with an alert definition.
type alertQuery struct {
```